src/duk_js_executor.c

```diff
@@ -2250,6 +2252,17 @@ DUK_INTERNAL void duk_js_execute_bytecode(duk_hthread *exec_thr) {
 		 * whenever a thread switch occurs by the DUK_HEAP_SWITCH_THREAD() macro.
 		 */
 #ifdef DUK_USE_INTERRUPT_COUNTER
+
+		if (thr->heap->dbg_processing) {
+			/* Don't interrupt if we're currently doing debug processing as
+			 * this may cause the debugger to be called recursively. Check required
+			 * for correct operation of throw intercept and other "exotic" halting
+			 * scenarios.
+			 */
+
+			goto skip_interrupt;
```

Comment:
In practice it should only be skipped for Eval command code though only, where we don't want the interrupt counter to run anyway?  That's what I thought, at least.